### PR TITLE
Support push down v2 filters

### PR DESCRIPTION
**Pull request by fbiville**
... internally propagate them as v1 filters, since these are already supported and easier to work with

### CODE REVIEW

File: common/src/main/scala/org/neo4j/spark/util/Neo4jUtil.scala [hunk 1]
```diff
@@ -260,7 +260,7 @@ object Neo4jUtil {
   def toParamValue(value: Any): Any = {
     value match {
       case date: java.sql.Date => date.toString
```
Comment (fbiville) [inline common/src/main/scala/org/neo4j/spark/util/Neo4jUtil.scala:251]: I'm wondering why this line did not need to change. Is it covered by tests?

File: common/src/main/scala/org/neo4j/spark/util/Neo4jUtil.scala [hunk 2]
```diff
@@ -155,23 +155,23 @@ object Neo4jUtil {
...
-    case date: java.sql.Date => convertFromSpark(date.toLocalDate, schema)
-    case timestamp: java.sql.Timestamp => convertFromSpark(timestamp.toInstant.atZone(ZoneOffset.UTC), schema)
-    case intValue: Int if schema != null && schema.dataType == DataTypes.DateType => convertFromSpark(DateTimeUtils
-      .toJavaDate(intValue), schema)
-    case longValue: Long if schema != null && schema.dataType == DataTypes.TimestampType => convertFromSpark(DateTimeUtils
-      .toJavaTimestamp(longValue), schema)
+  def convertFromSpark(value: Any, dataType: DataType = null): Value = value match {
+    case date: java.sql.Date => convertFromSpark(date.toLocalDate, dataType)
+    case timestamp: java.sql.Timestamp => convertFromSpark(timestamp.toLocalDateTime, dataType)
+    case intValue: Int if dataType != null && dataType == DataTypes.DateType => convertFromSpark(DateTimeUtils
```
Comment (fbiville) [inline common/src/main/scala/org/neo4j/spark/util/Neo4jUtil.scala]: we can simplify these checks to just rely on `==`

Comment (fbiville) [reply common/src/main/scala/org/neo4j/spark/util/Neo4jUtil.scala]: fixed in e5c9ad496cb5791b049f9426d3c97ee8cac85d19

**Review by conker84 — APPROVED** [review]
LGTM Thank you @fbiville !

**Pull request merged**